interface/wx/webrequest.h

```diff
@@ -0,0 +1,633 @@
+        header.
+
+        @param name Name of the header
+        @param value String value of the header
+    */
+    void SetHeader(const wxString& name, const wxString& value);
+
+    /**
+        Override the default temporary directory that may be used by the
+        session implementation, when required.
+    */
+    void SetTempDir(const wxString& dir);
+
+    /**
+        Returns the current temporary directory.
+
+        @see SetTempDir()
+    */
+    wxString GetTempDir() const;
+
+    /**
+        Returns the default session
+    */
+    static wxWebSession& GetDefault();
+
+    /**
+        Creates a new wxWebSession object.
+
+        @param backend
+            The backend web session implementation to use.
+
+        @return
+            The created wxWebSession
+    */
+    static wxWebSession* New(const wxString& backend = wxWebSessionBackendDefault);
+
+    /**
+        Allows the registering of new backend for wxWebSession.
+
+        backend can be used as an argument to New().
+
+        @param backend The name for the new backend to be registered under
+        @param factory A shared pointer to the factory which creates the appropriate backend.
+    */
+    static void RegisterFactory(const wxString& backend, wxSharedPtr<wxWebSessionFactory> factory);
+
+    /**
+        Allows to check if the specified backend is available at runtime.
+
+        Usually the default backend should always be available, but e.g. macOS
+        before 10.9 does not have the @c NSURLSession implementation available.
+    */
+    static bool IsBackendAvailable(const wxString& backend);
+};
+
+/**
+    @class wxWebSessionFactory
+
+    An abstract factory class for creation wxWebSession backends.
+
+    Each implementation of wxWebSession should have its own factory.
+
+    @since 3.1.2
+
+    @library{wxnet}
+    @category{net}
+
+    @see wxWebSession
+*/
+class wxWebSessionFactory
+{
+public:
+    /**
+        Creates a new web session object.
+    */
+    virtual wxWebSession* Create();
+};
+
+/**
+    @class wxWebRequestEvent
+
+    A web request event sent during or after server communication.
+
+    @since 3.1.2
+
+    @library{wxnet}
+    @category{net}
+
+    @see wxWebRequest
+*/
+class wxWebRequestEvent : public wxEvent
+{
+public:
+    wxWebRequestEvent();
+    wxWebRequestEvent(wxEventType type, int id, wxWebRequest::State state,
+        wxWebResponse* response = NULL, const wxString& errorDesc = "");
+
+    /**
+        Return the current state of the request
+    */
+    wxWebRequest::State GetState() const;
+
+    /**
+        The response with the state set to @c State_Complete or @c NULL for other
+        events.
+    */
+    wxWebResponse* GetResponse() const;
+
+    /**
+        A textual error description for a client side error
+        in case of @c State_Failed
+    */
+    const wxString& GetErrorDescription() const;
+
+    /**
+        Returns a file name to a temporary file containing the response data
```

Comment:
```suggestion
        Returns the name of a temporary file containing the response data
```
But is it only the file name or the complete path?